File: 3153/main.go
package main

import (
	"fmt"
	"runtime"
	"sync"
	"time"
)

func sumDigitDifferences(nums []int) int64 {
	res := int64(0)
	for nums[0] > 0 {
		cnt := make([]int, 10)
		for i := 0; i < len(nums); i++ {
			cnt[nums[i] % 10]++
			nums[i] /= 10
		}
		for i := 0; i < 10; i++ {
			res += int64(len(nums) - cnt[i]) * int64(cnt[i])
		}
	}
	return res / 2
}

func main() {
	numCPU := runtime.NumCPU()
    runtime.GOMAXPROCS(numCPU)
	fmt.Println(sumDigitDifferences([]int{13,23,12})) // 4
	foo()
}

func foo() {
	closed := make(chan struct{})
	go func() {
		time.Sleep(time.Second*6)
		close(closed)
		fmt.Println("goroutine end")
	}()
	// closed <- struct{}{}
	fmt.Printf("closed: %V\n", <-closed)

	time.Sleep(5 * time.Second)
	
	start := make(chan struct{})
	var wg sync.WaitGroup
	for i := 0; i < 3; i++ {
		wg.Add(1)
		fmt.Println(i)
		go func() {
			defer wg.Done()
			fmt.Printf("%V\n",<-start)
		}()
	}
	close(start)
	wg.Wait()

	select {
	case <-closed:
	case <-time.After(5 * time.Second):
		fmt.Println("timeout")
	}
}

File: 3153/main.rs
struct Solution {}

impl Solution {
    pub fn sum_digit_differences(nums: Vec<i32>) -> i64 {
        let mut nums = nums.clone();
        let mut res = 0i64;
        while nums[0] > 0 {
            let mut cnt: Vec<i32> = vec![0; 10];
            for i in 0..nums.len() {
                cnt[(nums[i as usize] % 10) as usize] += 1;
                nums[i as usize] /= 10;
            }
            for i in 0..10 {
                res += (nums.len() as i32 - cnt[i as usize]) as i64 * cnt[i as usize] as i64;
            }
        }
        return res / 2;
    }
}

fn main() {
    println!("{}", Solution::sum_digit_differences(vec![12,23,13]))  // 4
}